scripts/build.ts

```diff
@@ -39,3 +60,42 @@ function buildPackage() {
     encoding: "utf-8",
   });
 }
+
+function buildExtendedJSON() {
+  for (const [id, featureData] of Object.entries(data.features)) {
+    if (
+      Array.isArray(featureData.compat_features) &&
+      featureData.compat_features.length &&
+      featureData.status
+    ) {
+      featureData.status.by_compat_key = {};
+      for (const key of featureData.compat_features) {
+        featureData.status.by_compat_key[key] = getStatus(id, key);
+      }
+    }
+  }
+
+  if (!valid(data)) {
+    logger.error("Data failed schema validation. No JSON file written.");
+    process.exit(1);
+  }
+
+  fs.writeFileSync(
+    new URL("./web-features.extended.json", rootDir),
+    stringify(data),
+  );
+}
+
+function valid(data: any): boolean {
+  const valid = validate(data);
+  if (!valid) {
+    // TODO: turn on strictNullChecks, fix all the errors, and replace this with:
+    // const errors = validate.errors;
+    const errors = (valid as any).errors as DefinedError[];
```

Comment:
This is a bit ugly because ajv can't produce (useful) types without `strictNullChecks` set to `true` in `tsconfig.json`. Fixable, but it requires touching a dozen files not germane to this PR.